magento/magento2
# 11615: Fix UrlRewriteGenerator for anchor categories

A commit itself is pretty straightforward. Need to pass storeId param to category repository so resulted category has a correct url path.

### Description (*)
For not default stores adding a category for a product may result in wrong url rewrite for a product in category's anchor parent .
![result](http://dl4.joxi.net/drive/2019/01/31/0021/3892/1433396/96/558462029e.jpg)
### Fixed Issues (if relevant)

1.  [#11615](https://github.com/magento/magento2/issues/11615)

### Manual testing scenarios (*)
1. Enable this 
![config](http://dl3.joxi.net/drive/2019/01/30/0021/3892/1433396/96/901040f704.jpg)
2. Change url key of some anchor category in a specific (NOT DEFAULT) store 
![key](http://dl4.joxi.net/drive/2019/01/30/0021/3892/1433396/96/48c802a363.jpg)
3. add a product to a child category from a product form
![product](http://dl3.joxi.net/drive/2019/01/30/0021/3892/1433396/96/888ce74668.jpg)
Note that there is a correct result if a product is added from a category form.

### Contribution checklist (*)
